pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -695,10 +709,14 @@ func (o *MustGatherOptions) Run() error {
 				if quit {
 					return
 				}
-				defer queue.Done(pod)
-				if err := o.processNextWorkItem(ns.Name, pod.(*corev1.Pod)); err != nil {
-					errCh <- err
-				}
+				func() {
```

Comment:
Are we fixing an another problem?, or this change is related to context wiring?

Reply:
It is another problem really, although related to shutting down the queue on context cancelled.